server/rooms.ts

```diff
@@ -1935,13 +1933,19 @@ export const Rooms = {
 			} else if (playerOptions && playerOptions.hidden) {
 				privacySetter.add(playerOptions.user.id);
 			}
+			if (playerOptions) battle.checkForcedSettings(playerOptions.user);
+		}
+
+		// Special battles have modchat set to Player from the beginning
+		if (p1Special || battle.forcedSettings.modchat) {
+			room.settings.modchat = '\u2606';
 		}
 
 		if (privacySetter.size) {
-			if (battle.forcePublic) {
+			if (battle.forcedSettings.privacy) {
```

Comment:
Honestly, this all can.

Reply:
Should this be? It seems fine where it is (also, passing the options down seems messy and not worth it?).

Reply:
Never mind, I just went and did it.